typescript/packages/plugins/dragonswap/src/dragonswap.service.ts

```diff
@@ -0,0 +1,165 @@
+import { Tool } from "@goat-sdk/core";
+import { EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { ERC20_ABI } from "./abi/erc20abi";
+import { factoryAbi } from "./abi/factoryabi";
+import { quoterv2abi } from "./abi/quoterv2abi";
+import { swaprouterabi } from "./abi/swaprouterabi";
+import { POSITION_MANAGER_ABI } from "./abi/nftpositionmanager";
+import { DragonswapConfig } from "./dragonswap.plugin";
+import {
+    AddLiquidityParameters,
+    CollectFeesParameters,
+    ExactInputSingleParameters,
+    RemoveLiquidityParameters,
+} from "./parameters";
+
+export class DragonswapService {
+    constructor(private readonly config: DragonswapConfig) {}
+
+    @Tool({
+        name: "swap_exact_input_single_on_dragonswap",
+        description: "Swap tokens on Dragonswap using exact input amount",
+    })
+    async exactInputSingle(walletClient: EVMWalletClient, parameters: ExactInputSingleParameters) {
+        // Approve token if needed
```

Comment:
We spoke about this, approves happen thru the erc20 plugin. No need to add this or the erc20 abi :)

Reply:
gotcha yes , always forget . Will remove :)